stdlib/fetch.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3.6
+# -*- coding: utf-8 -*-
+
+import os
+import ntpath
+import stdlib
+import shutil
+import hashlib
+import requests
+import ftplib
+from urllib.parse import urlparse
+
+
+def fetch():
+    """Fetch the input data contained in the versionized argument ``fetch``.
+
+    The versionized argument ``fetch`` must be an array of dictionnaries. Each entry in the array is a data to fetch.
+    The dictionnary indicates how to retrieve the data and some optional parameters.
+
+    If the dictionnary contains the key:
+
+        * ``url``: the function :py:func:`.fetch_url` is called with the remaining values as arguments.
+        * ``file``: the function :py:func:`.fetch_file` is called with the remaining values as arguments.
+
+    If the dictionary contains both the key ``url`` and ``file``, the behaviour of the function is undefined.
```

Comment:
Are you sure this is the appropriate approach here ? We don't really benefit from undefined behaviour, so you might just wanna raise an exception instead to report the error